tests/unit/amo/components/TestCollection.js

```diff
@@ -1304,6 +1333,51 @@ describe(__filename, () => {
     );
   });
 
+  describe('onSortSelect', () => {
+    it.each([true, false])(
+      `calls router.push with expected pathname and query when a sort is selected and editing is %s`,
+      (editing) => {
+        const slug = 'some-slug';
+        const username = 'some-username';
+        const page = 2;
+        const sort = COLLECTION_SORT_NAME;
+        const clientApp = CLIENT_APP_FIREFOX;
+        const lang = 'en-US';
+
+        const { store } = dispatchClientMetadata({ clientApp, lang });
+        const router = createFakeRouter();
+        const preventDefaultSpy = sinon.spy();
+
+        const wrapper = renderComponent({
+          editing,
+          location: fakeRouterLocation({ query: { page, sort } }),
+          params: { username, slug },
+          router,
+          store,
+        });
+
+        // This emulates a user choosing an item from the sort select.
+        wrapper.instance().onSortSelect(
```

Comment:
Could we actually use `wrapper.simulate()` here? We should use `instance()` "parsimoniously".